### PR TITLE
Use up-to-date global coverage stats for browser versions

### PR DESCRIPTION
**Pull request by a-chabin**
I saw issues by @MonstraG (#512 & https://github.com/Fyrd/caniuse/issues/6811) about errors in browser usage statistics. 

## TL;DR
Now we will use stats that is updated dayly/weekly, not once a month (how it is implemented in the `browserlist.coverage()`)

### Before
image

### After
image

## Problem
image

- Site show stats from **[region-usage-json/alt-ww.json](https://github.com/Fyrd/caniuse/blob/main/region-usage-json/alt-ww.json)**  — updated **monthly**
- `browserslist.coverage()` use global stats from **[data.json](https://github.com/Fyrd/caniuse/blob/main/data.json)** — updated **daily/weekly**

## Changes
- [x] Use `usage_global` instead `alt-ww`
- [x] API returns `region: null` instead `alt-ww` by default
- [x] Add tests for `coverage` queries
- [x] Update API example in README.md
- [x] Improve readability (module `get-coverage.js`)

### CODE REVIEW

File: server/lib/get-coverage.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+export async function getBrowserVersionCoverage(id, version, region) {
+  let isGlobal = !region
+  if (isGlobal) {
+    return parseBrowserVersionCoverage(caniuseAgents[id].usage_global, version)
+  }
+
+  try {
+    if (region.includes('/')) {
```
Comment (ai) [inline server/lib/get-coverage.js:23]: Let’s use allow-list RegExp instead of prohibited `/` symbol.

I am afraid that we can miss something and will get hacked.

File: server/lib/get-coverage.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      throw new Error(`Invalid symbols in region name \`${region}\`.`)
+    }
+
+    let { default: regionData } = await import(
+      `caniuse-lite/data/regions/${region}.js`
+    )
+    return parseBrowserVersionCoverage(caniuseRegion(regionData)[id], version)
+  } catch (e) {
+    throw new Error(`Unknown region name \`${region}\`.`)
```
Comment (ai) [inline server/lib/get-coverage.js:32]:
```suggestion
    throw new Error(`Unknown region name \`${region}\``)
```

File: server/lib/get-coverage.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+export async function getBrowserVersionCoverage(id, version, region) {
+  let isGlobal = !region
+  if (isGlobal) {
+    return parseBrowserVersionCoverage(caniuseAgents[id].usage_global, version)
+  }
+
+  try {
+    if (region.includes('/')) {
+      throw new Error(`Invalid symbols in region name \`${region}\`.`)
```
Comment (ai) [inline server/lib/get-coverage.js:24]:
```suggestion
      throw new Error(`Invalid symbols in region name \`${region}\``)
```

File: server/lib/get-coverage.js (hunk 1 above)
Comment (a-chabin) [reply server/lib/get-coverage.js:23]: 👍👍👍

It's great that you noticed it.
I think we'll be fine with something like this:

```js
let isValidRegionName = /^(alt-)?[A-Za-z]{2}$/.test(region);
```

**Comment by a-chabin** [comment]
@ai 
I tested my changes a day later and found 2 bugs:

1. We don't need to set the region. I'll create new PR for client-side
2. Broken rounding. I'll fix it in current PR


image

File: server/lib/get-coverage.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+function parseBrowserVersionCoverage(stats, ver) {
+  let [lastVer] = Object.keys(stats).sort((a, b) => Number(b) - Number(a))
+
+  // If specific version coverage is missing, fall back to 'version zero'
+  let coverage = stats[ver] !== undefined ? stats[ver] : stats[lastVer]
+  return roundNumber(coverage)
+}
+
+function roundNumber(value) {
```
Comment (a-chabin) [inline server/lib/get-coverage.js:45]: Let's use shorter name
`roundNumber` → `round`

File: server/lib/get-browsers.js [hunk 5]
```diff
@@ -60,9 +57,7 @@ export async function getBrowsers(config, region) {
         coverage = null
       } else {
         name = caniuseAgents[id].browser
-        coverage = roundNumber(
-          Object.values(versions).reduce((a, b) => a + b, 0)
-        )
+        coverage = Object.values(versions).reduce((a, b) => a + b, 0)
```
Comment (a-chabin) [inline server/lib/get-browsers.js:60]: reduce() broke rounding
